pkg/transport/listener.go

```diff
@@ -114,7 +114,7 @@ func (info TLSInfo) Empty() bool {
 	return info.CertFile == "" && info.KeyFile == ""
 }
 
-func SelfCert(lg *zap.Logger, dirpath string, hosts []string, additionalUsages ...x509.ExtKeyUsage) (info TLSInfo, err error) {
+func SelfCert(lg *zap.Logger, dirpath string, hosts []string, validityYear uint, additionalUsages ...x509.ExtKeyUsage) (info TLSInfo, err error) {
```

Comment:
Can we return an error if `validityYear` is configured 0?

Reply:
done